internal/provider/fake_server_test.go

```diff
@@ -0,0 +1,251 @@
+package provider_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+
+	cloudexportpb "github.com/kentik/api-schema-public/gen/go/kentik/cloud_export/v202101beta1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+const cloudExportNotFound = -1
+
+type fakeCloudExportServer struct {
+	cloudexportpb.UnimplementedCloudExportAdminServiceServer
+	server *grpc.Server
+
+	url  string
+	done chan struct{}
+	t    testing.TB
+
+	data []*cloudexportpb.CloudExport
+}
+
+func newFakeCloudExportServer(t testing.TB, ces []*cloudexportpb.CloudExport) *fakeCloudExportServer {
+	return &fakeCloudExportServer{
+		done: make(chan struct{}),
+		t:    t,
+		data: ces,
+	}
+}
+
+func (s *fakeCloudExportServer) Start() {
+	l, err := net.Listen("tcp", "localhost:0")
```

Comment:
Is 0 really the correct port to listen on?

Reply:
When port is set to 0, system run server on random suitable port number.

Reply:
How does the API client used in test then know to which port to connect?